library/std/src/sync/once_lock.rs

```diff
@@ -294,30 +306,31 @@ impl<T> OnceLock<T> {
     }
 
     #[cold]
-    fn initialize<F, E>(&self, f: F) -> Result<(), E>
+    fn initialize<F, R>(&self, f: F) -> ControlFlow<R::Residual>
     where
-        F: FnOnce() -> Result<T, E>,
+        F: FnOnce() -> R,
+        R: Try<Output = T>,
     {
-        let mut res: Result<(), E> = Ok(());
```

Comment:
Might make sense to check for other uses of `res` in the relevant files, as that’s the name I use a lot.

Reply:
There is one other case in `Clone`, but I think it's fine as is. I just don't want to use `res` here since it could cause confusion as to whether it means "result" or "residual".